src/osp/Active/ActiveScene.h

```diff
@@ -152,16 +110,33 @@ class ActiveScene
         return m_registry.emplace<T>(ent, std::forward<Args>(args)...);
     }
 
+
+    /**
+     * Mark an entity for deletion
+     *
+     * @param ent [in] Entity to delete
+     */
+    void mark_delete(ActiveEnt ent)
+    {
+        m_registry.emplace_or_replace<ACompDelete>(ent);
```

Comment:
I feel like i've asked this before, but emplacement doesn't invalidate iterators to anything that's already in the registry right?

Reply:
Assuming so, probably want to elaborate on the thread safety in a comment.

Reply:
It only modifies the buffers associated with the templated component. Everything else stays untouched.